### ISSUE TITLE
Misc. bug: chat responses mode need change conver logic

### ISSUE THREAD

**Issue by wszgrcy** [ggml-org/llama.cpp#26394]
### Name and Version

version: 10216 (876a43211)
built with GNU 11.4.0 for Linux x86_64

### Operating systems

Linux

### Which llama.cpp modules do you know to be affected?

llama-server

### Command line

```shell
llama-server 
--model ./models/Qwen3.6-35B-A3B-Q8_0.gguf
```

### Problem description & steps to reproduce

- send
```
{
  "model": "gpt-5.4-nano",
  "input": [
    {
      "role": "developer",
      "content": "developer1"
    },
    {
      "role": "assistant",
      "content": [
        {
          "type": "output_text",
          "text": "assistant1"
        }
      ]
    },
    {
      "role": "user",
      "content": [
        {
          "type": "input_text",
          "text": "hello1"
        }
      ]
    }
  ],
  "store": false,
  "include": [
    "reasoning.encrypted_content"
  ],
  "reasoning": {
    "effort": "low"
  },
  "tools": [
   
  ],
  "tool_choice": "auto",
  "stream": true
}
```
throw error `Cannot determine type of 'item'`
I suspect the issue might be here: https://github.com/ggml-org/llama.cpp/blob/27209a598da5b9eb18cbf274e4e9b3367526b114/tools/server/server-chat.cpp#L108  
because the data sent does not contain type: 'message'  

```
   {
      "role": "assistant",
      "content": [
        {
          "type": "output_text",
          "text": "assistant1"
        }
      ]
    },
```

According to https://github.com/openai/openai-node/blob/d6dad64d3189256867ce5a6166b2bfc4cb8edc4a/src/resources/responses/responses.ts#L697 , it seems that `type: 'message'` can be omitted.

### First Bad Commit

none

### Relevant log output

<details>
<summary>Logs</summary>

```console
0.10.261.409 W srv    operator(): got exception: {"error":{"code":400,"message":"Cannot determine type of 'item'","type":"invalid_request_error"}}
```
</details>

**Comment by Burntgogi** [comment]
llama.cpp Responses API Compatibility Issue Verification

The issue reported in llama.cpp #26394 is still reproducible on build 10430 (commit 4c1a0af40) using Linux, CUDA, Qwen3.8-27B, and the `/v1/responses` endpoint.

The main reproduction case is an assistant history item that includes `role` and `content` but omits the optional `type: "message"` field. Before our local patch, the request failed with HTTP 400 and the following error:

`Cannot determine type of 'item'`

The error is raised in `server_chat_convert_responses_to_chatcmpl()` before model inference starts. This indicates that the problem is in **Responses API compatibility**, rather than in vision handling or mmproj.

Our local patch is intentionally minimal. Assistant output messages are accepted when `type` is either omitted or explicitly set to `"message"`. Explicitly invalid type values are still rejected with HTTP 400, preserving the existing validation behavior.

After applying the patch, the following cases were verified successfully:

* Text multi-turn Responses request: HTTP 200
* Image multi-turn Responses request: HTTP 200
* Chat Completions image request: HTTP 200

The image cases were included as regression checks to confirm that the change does not affect the existing vision request paths.

A regression case was added to `tests/test-chat.cpp`. The new test initially reproduced the reported exception, and then passed after the minimal change was applied. The final `test-chat` run completed with zero failures.

Relevant local files:

* `tools/server/server-chat.cpp`
* `tests/test-chat.cpp`

The corresponding git diff is attached

[llama-cpp-26394-b10430.patch](https://github.com/user-attachments/files/31240505/llama-cpp-26394-b10430.patch)

 for the exact local changes.

**Comment by EnricoFerrero64** [comment]
Confirming this on a current build, with one detail I did not see reported: **the rejection is asymmetric between roles**, which makes it easy to miss.

`role: "user"` without `type` is accepted. `role: "assistant"` without `type` is rejected. Same request shape, only the role differs.

Minimal reproduction — simpler than the original report, no `content` parts, no tools, no streaming:

```bash
# 400 — Cannot determine type of 'item'
curl -s -X POST http://HOST:PORT/v1/responses -H 'Content-Type: application/json' -d '{
  "model":"MODEL",
  "input":[{"role":"assistant","content":"B"},{"role":"user","content":"C"}],
  "max_output_tokens":20}'

# 200 — only the first role changed
curl -s -X POST http://HOST:PORT/v1/responses -H 'Content-Type: application/json' -d '{
  "model":"MODEL",
  "input":[{"role":"user","content":"B"},{"role":"user","content":"C"}],
  "max_output_tokens":20}'

# 200 — explicit type on the assistant item
curl -s -X POST http://HOST:PORT/v1/responses -H 'Content-Type: application/json' -d '{
  "model":"MODEL",
  "input":[{"type":"message","role":"assistant","content":"B"},{"role":"user","content":"C"}],
  "max_output_tokens":20}'

# 200 — same conversation through the other API
curl -s -X POST http://HOST:PORT/v1/chat/completions -H 'Content-Type: application/json' -d '{
  "model":"MODEL",
  "messages":[{"role":"user","content":"A"},{"role":"assistant","content":"B"},{"role":"user","content":"C"}],
  "max_tokens":20}'
```

| input item | result |
|---|---|
| `{"role":"user","content":...}` | 200 |
| `{"role":"assistant","content":...}` | **400** |
| `{"type":"message","role":"assistant",...}` | 200 |
| same conversation on `/v1/chat/completions` | 200 |

Reproduced on two separate binaries on the same host — stock `llama-server` (`build_info: b1-ece963f`) and a downstream fork — which behave identically, so this is not fork-specific.

On an embedding-only model the difference is especially clear, because the two failures happen at different stages: the untyped **assistant** item returns `400 Cannot determine type of 'item'` (rejected while parsing), while the untyped **user** item and the typed assistant item both get past parsing and only then return `500 the current context does not logits computation` from the model itself.

**Why it matters in practice.** Agent frameworks that replay a conversation tend to emit a *mixed* list: items they build from a tool-call result carry `type`, while synthetic assistant messages they insert themselves do not. With Hermes Agent v0.21.0 the trigger was an interruption placeholder:

```json
{"role": "assistant", "content": "[response interrupted]"}
```

Because the client retries, one malformed item costs the full round trip plus retries. Measured over one day of normal use: 11 rejected requests, on conversations as short as 3 items and as long as 173 — so it is not a long-context edge case.

Since `role` is present and unambiguous in all these cases, defaulting `type` to `"message"` for any role (not just `user`) would resolve it.
